### ISSUE TITLE
Pyod for Online and/or Incremental Learning

### ISSUE THREAD

**Issue by Lolom13** [yzhao062/pyod#191]
Hi,

I would like to know if among all the algorithms implemented in PyOD, some of them are thought for online outliers detection, or for Incremental Learning, which are two important problematics in some applications of Anomaly Detection.
(And if not, do you maybe know other open-source projects that have implemented such algorithms ?)

Thanks ! :)

**Comment by arnaudvl** [comment]
As far as I know, the implementations in PyOD are offline, including for methods which could be online such as [Loda](https://link.springer.com/article/10.1007/s10994-015-5521-0). Anomaly, adversarial and drift detection library [alibi-detect](https://github.com/SeldonIO/alibi-detect) does have a couple of online methods such as an online version of the [Mahalanobis distance](https://docs.seldon.io/projects/alibi-detect/en/latest/methods/mahalanobis.html) for tabular data (extended to work on a mixture of numerical and categorical features) and [Spectral Residuals](https://docs.seldon.io/projects/alibi-detect/en/latest/methods/sr.html) for time series data.

**Comment by yzhao062** [comment]
This is correct. No online learning or streaming setting is considered for now.

**Comment by selimfirat** [comment]
In addition to the first answer by @arnaudvl, PySAD might be the one you are looking for. PySAD directly focuses on anomaly detection on streaming data. PySAD provides a complete set of tools for streaming experimentation and various models for online anomaly detection. It also provides wrappers to integrate PyOD models to the streaming setting, called Integrators.

PySAD Repository: https://github.com/selimfirat/pysad
PySAD Documentation: https://pysad.readthedocs.io/en/latest/
